File: appMain/src/main/java/com/cartlc/trackbattery/data/PrefHelperBase.java
package com.cartlc.trackbattery.data;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by dug on 4/14/17.
 */

public class PrefHelperBase {

    protected SharedPreferences mPrefs;
    final protected Context mCtx;

    PrefHelperBase(Context ctx) {
        mCtx = ctx;
    }

    SharedPreferences getPrefs() {
        if (mPrefs == null) {
            mPrefs = mCtx.getSharedPreferences(getPrefFile(), 0);
        }
        return mPrefs;
    }

    String getPrefFile() {
        return mCtx.getPackageName() + "_preferences";
    }

    public String getString(String key, String defaultValue) {
        return getPrefs().getString(key, defaultValue);
    }

    public void setString(String key, String value) {
        SharedPreferences.Editor editor = getPrefs().edit();
        editor.putString(key, value);
        editor.commit();
    }

    public long getLong(String key, long defaultValue) {
        return getPrefs().getLong(key, defaultValue);
    }

    public void setLong(String key, long value) {
        SharedPreferences.Editor editor = getPrefs().edit();
        editor.putLong(key, value);
        editor.commit();
    }
}

File: appMain/src/main/java/com/cartlc/tracker/fresh/model/pref/PrefHelperBase.kt
/**
 * Copyright 2017-2021, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.fresh.model.pref

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by dug on 4/14/17.
 */

open class PrefHelperBase constructor(protected val ctx: Context) {

    protected val prefs: SharedPreferences by lazy {
        ctx.getSharedPreferences(prefFile, 0)
    }

    private val prefFile: String
        get() = ctx.packageName + "_preferences"

    fun getString(key: String, defaultValue: String?): String? {
        return prefs.getString(key, defaultValue)
    }

    fun setString(key: String, value: String?) {
        val editor = prefs.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return prefs.getLong(key, defaultValue)
    }

    fun setLong(key: String, value: Long) {
        val editor = prefs.edit()
        editor.putLong(key, value)
        editor.apply()
    }

    fun getInt(key: String, defaultValue: Int): Int {
        return prefs.getInt(key, defaultValue)
    }

    fun setInt(key: String, value: Int) {
        val editor = prefs.edit()
        editor.putInt(key, value)
        editor.apply()
    }

    open fun clearAll() {
        val editor = prefs.edit()
        editor.clear()
        editor.commit()
    }

}
